python_modules/dagster/dagster/_core/instance/config.py

```diff
@@ -600,3 +603,74 @@ def dagster_instance_config_schema() -> Mapping[str, Field]:
         ),
         "concurrency": get_concurrency_config(),
     }
+
+
+class PoolGranularity(Enum):
+    OP = "op"
+    RUN = "run"
+
+
+@record
+class PoolConfig:
+    pool_granularity: Optional[PoolGranularity]
+    default_pool_limit: Optional[int]
+    op_granularity_run_buffer: Optional[int]
+
+
+@record
+class ConcurrencyConfig:
+    pool_config: PoolConfig
+    run_queue_config: Optional["RunQueueConfig"]
+
+    @staticmethod
+    def from_concurrency_settings(
+        concurrency_settings: Mapping[str, Any],
+        run_coordinator_run_queue_config: Optional["RunQueueConfig"],
+    ) -> "ConcurrencyConfig":
+        from dagster._core.run_coordinator.queued_run_coordinator import RunQueueConfig
+
+        run_settings = concurrency_settings.get("runs", {})
+        pool_settings = concurrency_settings.get("pools", {})
+        pool_granularity_str = pool_settings.get("granularity")
+        if run_coordinator_run_queue_config:
+            run_queue_config = RunQueueConfig(
+                max_concurrent_runs=run_settings.get(
+                    "max_concurrent_runs", run_coordinator_run_queue_config.max_concurrent_runs
+                ),
+                tag_concurrency_limits=run_settings.get(
+                    "tag_concurrency_limits",
+                    run_coordinator_run_queue_config.tag_concurrency_limits,
+                ),
+                max_user_code_failure_retries=run_coordinator_run_queue_config.max_user_code_failure_retries,
+                user_code_failure_retry_delay=run_coordinator_run_queue_config.user_code_failure_retry_delay,
+                should_block_op_concurrency_limited_runs=bool(pool_settings)
+                or run_coordinator_run_queue_config.should_block_op_concurrency_limited_runs,
+                op_concurrency_slot_buffer=pool_settings.get(
+                    "op_granularity_run_buffer",
+                    run_coordinator_run_queue_config.op_concurrency_slot_buffer,
+                ),
+            )
+        else:
+            run_queue_config = None
+
+        pool_granularity = PoolGranularity(pool_granularity_str) if pool_granularity_str else None
+        if (
+            not pool_granularity
+            and run_coordinator_run_queue_config
+            and run_coordinator_run_queue_config.should_block_op_concurrency_limited_runs
+        ):
+            # if this was explicitly configured in the run coordinator config, we should default to op granularity
```

Comment:
ah ok i see, this is eventually going to supplant the run coordinator config then?

Reply:
Right now, I've only moved over `max_concurrent_runs` and the `tag_concurrency_limits`... essentially only the settings that control limits.

I was leaning towards leaving some things like the number of workers and dequeue interval seconds and retry settings in the run queue settings, but I'm open to moving them over just for simplicity's sake...